File: LP1/java/DPR.java

public class DPR {
    private int dataID;
    private String nama;
    private String bidang;
    private String partai;

    public DPR() {
        this.dataID = -1;
        this.nama = "";
        this.bidang = "";
        this.partai = "";
    }

    public DPR(int id, String nama, String bidang, String partai) {
        this.dataID = id;
        this.nama = nama;
        this.bidang = bidang;
        this.partai = partai;

    }

    // method setter getter
    public int getID() {
        return this.dataID;
    }

    public void setID(int ID) {
        this.dataID = ID;
    }

    public String getName() {
        return this.nama;
    }

    public void setName(String name) {
        this.nama = name;
    }

    public String getBidang() {
        return this.bidang;
    }

    public void setBidang(String bidang) {
        this.bidang = bidang;
    }

    public String getPartai() {
        return this.partai;
    }

    public void setPartai(String partai) {
        this.partai = partai;
    }

}

File: LP1/cpp/DPR.cpp
#include <iostream>
#include <string>

// using std namespace
using namespace std;

class DPR
{
private:
    // berisi atribut private karena ini data penting objeknya
    int dataID;
    string nama;
    string namaBidang;
    string namaPartai;

public:
    // construktor pertama yang isi nya defaul tanpa parameter atribut
    DPR()
    {
        this->dataID = -1;
        this->nama = "";
        this->namaBidang = "";
        this->namaPartai = "";
    }
    // versi ada parameter atribut biar langsung duar jadi lengkap
    DPR(int dataID, string nama, string namaBidang, string namaPartai)
    {
        this->dataID = dataID;
        this->nama = nama;
        this->namaBidang = namaBidang;
        this->namaPartai = namaPartai;
    }

    // method create update, delete

    // getter setter
    // ambil data dari private lalu return
    // set atribut private dengan parameter
    int getID()
    {
        return this->dataID;
    }
    void setID(int ID)
    {
        this->dataID = ID;
    }

    string getNama()
    {
        return this->nama;
    }
    void setNama(string nama)
    {
        this->nama = nama;
    }

    string getBidang()
    {
        return this->namaBidang;
    }
    void setBidang(string bidang)
    {
        this->namaBidang = bidang;
    }

    string getPartai()
    {
        return this->namaPartai;
    }
    void setPartai(string partai)
    {
        this->namaPartai = partai;
    }
    ~DPR(){

    };
};